final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,274 @@
+import argparse
+import html
+import json
+import feedparser
+import logging
+import urllib.request
+import sys
+import re
+from typing import Tuple, List
+
+import single_article
+import custom_error
+
+VERSION = 1.0
+LOG_FILE_NAME = 'app.log'
+
+
+def parse_rss(rss_url: str) -> feedparser.FeedParserDict:
+    """Parse the rss"""
+    return feedparser.parse(rss_url)
+
+
+def get_source(parsed: feedparser.FeedParserDict) -> dict:
+    """Returns dictionary with feed information."""
+    feed = parsed['feed']
+    return {
+        'link': feed['link'],
+        'title': feed['title'],
+        'subtitle': feed['subtitle'],
+    }
+
+
+def find_links_in_article(article: feedparser.FeedParserDict) -> list:
+    """Finds links in the article"""
+    image_search = re.compile(r'<\s*img [^>]*src="([^"]+)')
+    img = image_search.findall(article['summary'])
+
+    all_links = []
+    for link in article['links']:
+        if link['href'] not in img:
+            all_links.append(link['href'])
+    all_links += img
+
+    return all_links
+
+
+def unescape(text_to_unescape: str) -> str:
+    """Unescape text"""
+    return html.unescape(text_to_unescape)
+
+
+def get_articles(parsed: feedparser.FeedParserDict, limit: int) -> Tuple[List[single_article.SingleArticle], int]:
+    """Returns list with articles."""
+    articles = []
+    feed = parsed['feed']
+    entries = parsed['entries']
+    logging.info(f'There is {len(entries)} entries')
+
+    no_tags = re.compile('<.*?>')
+
+    if limit is None:
+        limit = len(entries)
+
+    try:
+        for index, entry in enumerate(entries):
+            if index == limit:
+                break
+
+            links_in_article = find_links_in_article(entry)
+
+            if 'published' not in entry:
+                limit -= 1
+                continue
```

Comment:
A question:

Could you please elaborate in what situation do you need this check?

Reply:
http://rss.cnn.com/rss/edition_world.rss bottom artricles don't have date of publication

Reply:
And so why do you lower limit by 1?

Reply:
To cout how much articles was printed

Reply:
So, according to this code, for example if user specifies `--limit 3` and there will be 2 articles with no publication date, he/she will not get any piece of news.

If you want to count how much articles you print it is better to just check length of `articles` list after you finish appending elements to it.